tempodb/tempodb.go

```diff
@@ -79,7 +79,7 @@ type Writer interface {
 }
 
 type Reader interface {
-	Find(ctx context.Context, tenantID string, id encoding.ID) ([]byte, FindMetrics, error)
+	Find(ctx context.Context, tenantID string, id encoding.ID, blockStart []byte, blockEnd []byte) ([]byte, FindMetrics, error)
```

Comment:
take uuid.UUID's instead of []bytes for blockStart/End

Reply:
I changed this to string, because otherwise we would have to pass `uuid` in the grpc message `tempopb.TraceByIDRequest`, which doesn't exist yet. Or, we can change signature for the `FindTraceByID()` function.

Reply:
I think string is nicer, it adds another layer of checks in `tempodb.go` but we should anyway have checks there because we're trying to keep it independent of the rest of the codebase (modules and stuff).

But if we have strong opinions to pass uuid I can see what we can do to change to that.